platforms/chibios/drivers/analog.c

```diff
@@ -240,6 +261,103 @@ __attribute__((weak)) adc_mux pinToMux(pin_t pin) {
         case C5:  return TO_MUX( ADC_CHANNEL_IN15, 0 );
         // STM32F103x[C-G] in 144-pin packages also have analog inputs on F6...F10, but they are on ADC3, and the
         // ChibiOS ADC driver for STM32F1xx currently supports only ADC1, therefore these pins are not usable.
+#elif defined(STM32L4XX)
+        case A0: return TO_MUX( ADC_CHANNEL_IN5,  0 ); // Can also be ADC2 in some cases
+        case A1: return TO_MUX( ADC_CHANNEL_IN6,  0 ); // Can also be ADC2 in some cases
+        case A2: return TO_MUX( ADC_CHANNEL_IN7,  0 ); // Can also be ADC2
+        case A3: return TO_MUX( ADC_CHANNEL_IN8,  0 ); // Can also be ADC2
+        case A4: return TO_MUX( ADC_CHANNEL_IN9,  0 ); // Can also be ADC2
+        case A5: return TO_MUX( ADC_CHANNEL_IN10, 0 ); // Can also be ADC2
+        case A6: return TO_MUX( ADC_CHANNEL_IN11, 0 ); // Can also be ADC2
+        case A7: return TO_MUX( ADC_CHANNEL_IN12, 0 ); // Can also be ADC2
+        case B0: return TO_MUX( ADC_CHANNEL_IN15, 0 ); // Can also be ADC2
+        case B1: return TO_MUX( ADC_CHANNEL_IN16, 0 ); // Can also be ADC2
+        case C0: return TO_MUX( ADC_CHANNEL_IN1,  0 ); // Can also be ADC2 or ADC3
+        case C1: return TO_MUX( ADC_CHANNEL_IN2,  0 ); // Can also be ADC2 or ADC3
+        case C2: return TO_MUX( ADC_CHANNEL_IN3,  0 ); // Can also be ADC2 or ADC3
+        case C3: return TO_MUX( ADC_CHANNEL_IN4,  0 ); // Can also be ADC2 or ADC3
+        case C4: return TO_MUX( ADC_CHANNEL_IN13, 0 ); // Can also be ADC2
+        case C5: return TO_MUX( ADC_CHANNEL_IN14, 0 ); // Can also be ADC2
+#    if STM32_ADC_USE_ADC3
```

Comment:
Technically the more important check here is `STM32_HAS_GPIOF` (without `GPIOF` this code won't compile at all); checking just `STM32_ADC_USE_ADC3` relies on the fact that chips without `GPIOF` also don't have ADC3, therefore enabling `STM32_ADC_USE_ADC3` would break the build anyway due to a check in ChibiOS.

Reply:
Will implement that check in here to be safe and allow proper compilation 👍